src/processTargets/modifiers/SmallPaintStage.ts

```diff
@@ -0,0 +1,66 @@
+import { zip } from "lodash";
+import { Target } from "../../typings/target.types";
+import {
+  ContainingScopeModifier,
+  EveryScopeModifier,
+} from "../../typings/targetDescriptor.types";
+import { ProcessedTargetsContext } from "../../typings/Types";
+import { ModifierStage } from "../PipelineStages.types";
+import { TokenTarget } from "../targets";
+import { NonWhitespaceSequenceStage } from "./scopeTypeStages/RegexStage";
+import { InteriorOnlyStage } from "./InteriorStage";
+
+export type NonWhitespaceOrQuoteSequenceModifier = (
+  | ContainingScopeModifier
+  | EveryScopeModifier
+) & {
+  scopeType: { type: "nonWhitespaceOrQuoteSequence" };
+};
+
+/**
+ * Intersection of NonWhitespaceSequenceStage and InteriorOnlyStage
+ * Expand the target until reaching a white space or surrounding pair.
+ * If there is no surrounding pair defaults to the non white space sequence
+ */
+export default class SmallPaintStage implements ModifierStage {
+  constructor(private modifier: NonWhitespaceOrQuoteSequenceModifier) {}
+
+  run(context: ProcessedTargetsContext, target: Target): Target[] {
+    const paintStage = new NonWhitespaceSequenceStage({
+      type: "containingScope",
+      scopeType: { type: "nonWhitespaceSequence" },
+    });
+
+    const surroundingPairStage = new InteriorOnlyStage({
+      type: "interiorOnly",
+    });
+
+    const paintTargets = paintStage.run(context, target);
+
+    try {
+      const pairTargets = surroundingPairStage.run(context, target);
+      return zip(paintTargets, pairTargets).map(([paintTarget, pairTarget]) => {
```

Comment:
why are we zipping here?

Reply:
Because the stages return an array but that will change for surrounding pair